### PR TITLE
Fix flaky test:test_sample_multinomial

### LINKED ISSUE #10838: Flaky test_random.test_sample_multinomial
Link to the failed build: http://jenkins.mxnet-ci.amazon-ml.com/blue/organizations/jenkins/incubator-mxnet/detail/PR-10533/34/pipeline
```
======================================================================

FAIL: test_random.test_sample_multinomial

----------------------------------------------------------------------

Traceback (most recent call last):

  File "C:\Anaconda3\envs\py2\lib\site-packages\nose\case.py", line 197, in runTest

    self.test(*self.arg)

  File "C:\jenkins_slave\workspace\ut-python-gpu\tests\python\unittest\common.py", line 157, in test_new

    orig_test(*args, **kwargs)

  File "C:\jenkins_slave\workspace\ut-python-gpu\tests\python\unittest\test_random.py", line 400, in test_sample_multinomial

    mx.test_utils.assert_almost_equal(freq, x[i], rtol=0.20)

  File "C:\jenkins_slave\workspace\ut-python-gpu\pkg_vc14_gpu\python\mxnet\test_utils.py", line 493, in assert_almost_equal

    raise AssertionError(msg)

AssertionError: 

Items are not equal:

Error 10000000317344786.000000 exceeds tolerance rtol=0.200000, atol=0.000000.  Location of maximum error:(4,), a=0.000200, b=0.000000

 a: array([ 0.3922,  0.3044,  0.2064,  0.0968,  0.0002])

 b: array([ 0.40000001,  0.30000001,  0.2       ,  0.1       ,  0.        ], dtype=float32)

-------------------- >> begin captured logging << --------------------

common: INFO: Setting test np/mx/python random seeds, use MXNET_TEST_SEED=22688242 to reproduce.

--------------------- >> end captured logging << ---------------------

```

### PR DESCRIPTION
**Pull request by anirudh2290**
## Description ##

The test is used to obtain 5000 samples from `random.multinomial` op when provided a probability ndarray as input. It calculates count of the number of samples of each index (in input) and divides it by total samples to obtain the probability. This computed probability is then compared to the actual probability that is provided to the op. To make this test completely non flaky we would have to obtain infinite number of samples. To alleivate the pain caused by flakiness of this test for now, I have increased the number of samples to 10k, increased atol to 0.1 and also added a retry for the test if it fails. Hopefully, these three actions should reduce the flakiness of the test by a huge extent. I tested for 2000 runs on my machine without failure.

Fixes #10838 

@haojin2  @asitstands 

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by anirudh2290** [comment]
@eric-haibin-lin please help with review and merge.

File: tests/python/unittest/test_random.py [hunk 1]
```diff
@@ -358,6 +358,7 @@ def test_parallel_random_seed_setting_for_context():
         for i in range(1, len(samples_sym)):
             assert same(samples_sym[i - 1], samples_sym[i])
 
+@retry(5)
 @with_seed()
 @unittest.skip("Flaky test: https://github.com/apache/incubator-mxnet/issues/11487")
```
Comment (eric-haibin-lin) [inline tests/python/unittest/test_random.py]: Remove this line?

**Comment by szha** [comment]
> To make this test completely non flaky we would have to obtain infinite number of samples.
You don't need infinite number of samples to establish a confidence level.

**Comment by anirudh2290** [comment]
@szha what i am saying is there is still a very small chance that the test may fail. For practical purposes it should be fine.

**Comment by szha** [comment]
@anirudh2290 what's the current estimate for failure rate?

**Comment by anirudh2290** [comment]
@szha Intuitively, increasing the number of samples should lead to convergence to the probability vector. I cannot give you an estimate for the failure rate but  it should decrease with the increased number of samples. This in addition to the increased number of retries and atol should reduce the chances of a failure.

**Comment by anirudh2290** [comment]
is this pr good to merge ?

**Pull request merged**